### PR TITLE
Path matcher bug

### PR DESCRIPTION
**Pull request by manorlh**
path matcher faild to match to correct schema key when the routes are:
/api/v1/sar/reports/:requestId
/api/v1/sar/reports/bulk

(in yaml)

for request to path /api/v1/sar/reports/bulk, it match /api/v1/sar/reports/:requested.

### CODE REVIEW

**Comment by manorlh** [comment]
@kibertoad please review:)

**Comment by manorlh** [comment]
I will try to do it better, to support more cases

File: lib/helpers/schema-utils.js [hunk 1]
```diff
@@ -35,14 +35,20 @@ module.exports.getSchemaByFilesPath = (apiDefinitionsPath, buildSchemaOptions) =
...
 module.exports.getValidatorByPathMethodAndCode = (schema, request, response) => {
-  const route = pathMatcher(schema, request.path);
+  const route = pathMatcher(schema, request.path, request.method);
   return get(schema, `${route}.${request.method}.responses.${response.status}`);
 };
 
 module.exports.pathMatcher = pathMatcher;
-function pathMatcher(routes, path) {
+function pathMatcher(routes, path, method) {
   return Object
```
Comment (kobik) [inline lib/helpers/schema-utils.js:44]: lines 44-51 should be extracted and run once per routes, no need to sort on each call

File: lib/helpers/schema-utils.js [hunk 2]
```diff
@@ -35,14 +35,20 @@ module.exports.getSchemaByFilesPath = (apiDefinitionsPath, buildSchemaOptions) =
...
     .keys(routes)
+    .sort((firstEl, secondEl) => {
+      const firstResult = firstEl.split(':').map((value, index) => `${index}`).join('');
```
Comment (kobik) [inline lib/helpers/schema-utils.js:47]: this sort doesn't take into account different positions that `:` can be, i.e

```js
> '/aaaa/:bbbb'.split(':').map((value, index) => `${index}`).join('')
'01'
> '/aaaa:/bbbb'.split(':').map((value, index) => `${index}`).join('')
'01'
```
this logic takes into account the path segment that `:` was found in and not its absolute position
```js
> '/aaaa/:bbbb'.split('/').map((seg) => seg.includes(':')).map((seg, i, arr) => seg * Math.pow(10, arr.length - i)).reduce((sum, seg) => sum + seg, 0)
10
> '/aaaa:/bbbb'.split('/').map((seg) => seg.includes(':')).map((seg, i, arr) => seg * Math.pow(10, arr.length - i)).reduce((sum, seg) => sum + seg, 0)
100
```

anyway, the weight calculation should be extracted to a function

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,24 @@
 # Master
 
+# 2.2.4 - 15 March, 2021
+- a fix bug: route matcher
```
Comment (kobik) [inline CHANGELOG.md:4]:
```md
## Fixes
* Prioritize exact path segment matching
 ```

**Review by kobik — CHANGES_REQUESTED** [review]
thanks @manorlh , please see my comments

**Comment by manorlh** [comment]
@kobik Its ready for another check 
thanks

**Review by kobik — APPROVED** [review]

**Pull request merged**